wcomponents-core/src/main/java/com/github/bordertech/wcomponents/WMultiFileWidget.java

```diff
@@ -855,7 +869,9 @@ private boolean selectionsEqual(final List<?> list1, final List<?> list2) {
 	 * @param fileId the file id that has been uploaded successfully
 	 */
 	private void setFileUploadRequestId(final String fileId) {
-		getOrCreateComponentModel().fileUploadRequestId = fileId;
+		if (getFileUploadRequestId() == null || !getFileUploadRequestId().equals(fileId)) {
```

Comment:
Same as up there ^ `getFileUploadRequestId() == null` and `fileId` is null - do we still need the component model?